boards/shields/adafruit_2_8_tft_touch_v2/boards/nrf52840_pca10056.conf

```diff
@@ -0,0 +1 @@
+CONFIG_SPI_3=y
```

Comment:
Why not defining this in board ?
Makes sense to enable arduino related peripherals directly in boards if connector is present

Reply:
I do not think so. The board in default configuration does not need it and a board with shield is again a different device. For example, a shield with I2C and GPIO do not event need a SPI, less code size and energy consumption.

Reply:
What I propose is the following in board `Kconfig.defconfig`:
```
if SPI

config SPI_3
    default y

endif
```

So, we don't blindly enable subsystem not requested (which I agree is really non optimised). But we potentially enable a non requested instance of a peripheral already enabled by the application. This is indeed a drawback, but I consider it rather minor, and it could be easily disabled in an end user application prf.cnf that wants to optimize its footprint/consumption.

Advantage is that shields could be supported on much more boards w/ minimal modifications. And I think that this is what upstream zephyr repo should provide: maximum application supports on boards with minimum modifications. I think we can afford minor non optimized configuration( like extra device instance) to reach that goal.

Reply:
I removed it.

Reply:
@jfischer-phytec-iot my comment is after the fact but this file should have stayed in this PR.  This is the file that configures the nrf52840_pca10056's hardware requirements for the adafruit_2_8_tft_touch_v2 shield.  It is only applied when building with the adafruit_2_8_tft_touch_v2 specified as a shield so nothing blind about it.   If you're keen I'd make another PR to add it.

Reply:
No, this was already discussed with @erwango and have to be done by the board configuration.

Reply:
@nixward see [For software compatibility, boards must also provide a configuration matching their supported connectors.](https://docs.zephyrproject.org/latest/guides/porting/shields.html?highlight=shield#board-compatibility)

Reply:
@jfischer-phytec-iot thanks for correction/info!